src/elements/form/Control.ts

```diff
@@ -35,13 +35,13 @@ import { Helpers } from './../../utils/Helpers';
                     <!--Text-based Inputs-->
                     <!--TODO prefix/suffix on the control-->
                     <div class="novo-control-input-container novo-control-input-with-label" *ngSwitchCase="'textbox'">
-                        <input *ngIf="control.type !== 'number'" [formControlName]="control.key" [id]="control.key" [type]="control.type" [placeholder]="control.placeholder" (input)="emitChange($event)" [maxlength]="control.maxlength" (focus)="handleFocus($event)" (blur)="handleBlur($event)">
+                        <input *ngIf="control.type !== 'number'" [formControlName]="control.key" [id]="control.key" [type]="control.type" [placeholder]="control.placeholder" (input)="emitChange($event)" (keypress)="checkMaxLength($event)" [maxlength]="control.maxlength" (focus)="handleFocus($event)" (blur)="handleBlur($event)">
                         <input *ngIf="control.type === 'number'" [formControlName]="control.key" [id]="control.key" [type]="control.type" [placeholder]="control.placeholder" (keypress)="restrictKeys($event)" (input)="emitChange($event)" [maxlength]="control.maxlength" (focus)="handleFocus($event)" (blur)="handleBlur($event)" step="any">
                         <label class="input-label" *ngIf="control.subType === 'currency'">{{control.currencyFormat}}</label>
                         <label class="input-label" *ngIf="control.subType === 'percentage'">%</label>
                     </div>
                     <!--TextArea-->
-                    <textarea *ngSwitchCase="'text-area'" [name]="control.key" [attr.id]="control.key" [placeholder]="control.placeholder" [formControlName]="control.key" (input)="resizeTextArea($event)" (input)="emitChange($event)" [maxlength]="control.maxlength"></textarea>
+                    <textarea *ngSwitchCase="'text-area'" [name]="control.key" [attr.id]="control.key" [placeholder]="control.placeholder" [formControlName]="control.key" (input)="resizeTextArea($event)" (keypress)="checkMaxLength($event)" (input)="emitChange($event)" [maxlength]="control.maxlength"></textarea>
```

Comment:
Don't need keypress, you have the call inside emitChange, so you are calling it twice. Can remove the keypress event listenter.